src/bin/deku_node.ml

```diff
@@ -53,6 +53,17 @@ let handle_block_by_hash =
     (fun _update_state request ->
       let block = Flows.find_block_by_hash (Server.get_state ()) request.hash in
       Ok block)
+let handle_block_by_level =
+  handle_request
+    (module Networking.Block_by_level_spec)
+    (fun _update_state request ->
+      let state = Server.get_state () in
+      let block =
+        List.find_opt
+          (fun block ->
+            Int64.equal block.Protocol.Block.block_height request.level)
+          state.applied_blocks in
```

Comment:
Fine for now, but it's kind of sad that we can't just index into the list

Reply:
I believe the problem isn't indexing this set of blocks, but that the `state.applied_blocks` is unbounded currently. I made a TODO related to this in `./node/state.ml` and referenced the relevant issue (#535)

Reply:
When we have an indexer, we can discard any block that is below the most recently finalized blocks (at least, this is the case once we have deterministic consensus with tendermint).